app/src/main/cpp/skyline/services/timesrv/time_shared_memory.h

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2021 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#pragma once
+
+#include <common.h>
+#include <common/uuid.h>
+#include <kernel/types/KEvent.h>
+#include <kernel/types/KSharedMemory.h>
+#include <services/timesrv/common.h>
+
+namespace skyline::service::timesrv::core {
+    struct TimeSharedMemoryLayout;
+
+    /**
+     * @brief TimeSharedMemory holds context data about clocks in a double buffered format
+     */
+    class TimeSharedMemory {
+      private:
+        std::shared_ptr<kernel::type::KSharedMemory> kTimeSharedMemory;
+        TimeSharedMemoryLayout *timeSharedMemory;
+
+      public:
+        TimeSharedMemory(const DeviceState &state);
+
+        std::shared_ptr<kernel::type::KSharedMemory> GetSharedMemory() {
+            return kTimeSharedMemory;
+        }
+
+        /**
+         * @brief Fills in the steady clock section of shmem, the current time is subtracted from baseTimePoint to workout the offset
+         */
+        void SetupStandardSteadyClock(UUID rtcId, TimeSpanType baseTimePoint);
+
+        void SetSteadyClockRawTimePoint(TimeSpanType timePoint);
+
+        void UpdateLocalSystemClockContext(const SystemClockContext &context);
+
+        void UpdateNetworkSystemClockContext(const SystemClockContext &context);
+
+        void SetStandardUserSystemClockAutomaticCorrectionEnabled(bool enabled);
+    };
+
+    /**
+     * @brief Base class for callbacks that run after a system clock context is updated
+     */
+    class SystemClockContextUpdateCallback {
+      private:
+        std::list<std::shared_ptr<kernel::type::KEvent>> operationEventList; //!< List of KEvents to be signalled when this callback is called
```

Comment:
Same as above with just pluralizing it rather than tagging along `List` as a prefix, there might've been instances of this I've missed so do check for them